### ISSUE TITLE
Trailing dash in urls being dropped in sms - some android devices

### ISSUE THREAD

**Issue by irfaan** [dylang/shortid#96]
Just a PSA - I have noticed that, when texting a url with a trailing dash, some android devices are dropping the trailing dash when parsing urls in text messages. It seems to only happen when text follows a url with a trailing dash.

![screen shot 2017-05-11 at 12 04 44 pm](https://cloud.githubusercontent.com/assets/1435018/25964573/8b9b9ba4-3642-11e7-9139-56bbe82f7f52.png)

**Comment by tsriram** [comment]
Happened to see the same issue today. As a quick fix I'm going to use `~` instead of `-` in the set of characters and hope it doesn't cause any problem :)

**Comment by irfaan** [comment]
@tsriram Great idea. Will do the same. Thank you!

Section 2.3 of [RFC 3986](http://www.ietf.org/rfc/rfc3986.txt) confirms that ``~`` is url safe.

**Comment by tsriram** [comment]
@irfaan yeah, some (if not all) of Android devices drop tilde too. I'm currently doing a brute force way of generating shortid repeatedly till I get one that doesn't end with a special character. Not at all the best way but I was running out of time to dig into it. And can't have _broken_ URLs sent to users!

**Comment by GoNode5** [comment]
not sure but if you replace - with %2D

**Comment by bammoo** [comment]
```js
var shortid = require('shortid')
var shortidNotEndWithSpecialCharacter = function () {
  var id
  do {
    id = shortid.generate();
  }
  while(id.slice(-1) === '-' || id.slice(-1) === '_');
  return id;
}
```

**Comment by ai** [comment]
Removing IDS with last `-` is bad for uniformity. It is better to do `shortid.generate().replace(/-/g, '~')` or use [nanoid](https://github.com/ai/nanoid)

**Comment by ai** [comment]
Change alphabet by:

```js
shortid.characters(0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ~_)
```

Or use [Nano ID](https://github.com/ai/nanoid) which use `~` instead of `-`.

**Comment by irfaan** [comment]
@ai happens with tilde (~), too: https://github.com/dylang/shortid/issues/96#issuecomment-306293440
